### PR TITLE
feature: Enable fewer braces for Scala 3 nightly versions

### PR DESCRIPTION
**Pull request by tgodzik**
Not adding any tests yet, since the feature is still not approved via SIP, but we will not get any errors thanks to this.

### CODE REVIEW

File: sbt-metals/src/sbt-test/sbt-metals/semanticdb/build.sbt [hunk 1]
```diff
@@ -30,7 +30,7 @@ lazy val b = project
 lazy val c = project
   .in(file("c"))
   .settings(
-    scalaVersion := "2.12.8", // latest scalameta doesn't publish for ~2.12.7
+    scalaVersion := "2.12.9", // latest scalameta doesn't publish for ~2.12.8
```
Comment (kpodsiad) [inline sbt-metals/src/sbt-test/sbt-metals/semanticdb/build.sbt:33]: Don't they? https://mvnrepository.com/artifact/org.scalameta/semanticdb-scalac - 4.5.13 is available for 2.12.8

**Review by dos65 — APPROVED** [review]
:+1:

File: sbt-metals/src/sbt-test/sbt-metals/semanticdb/build.sbt (hunk 1 above)
Comment (tanishiking) [reply sbt-metals/src/sbt-test/sbt-metals/semanticdb/build.sbt:33]: https://github.com/scalameta/metals/pull/4463

File: metals/src/main/scala/scala/meta/internal/decorations/SemanticdbTreePrinter.scala [hunk 2]
```diff
@@ -86,6 +86,14 @@ class SemanticdbTreePrinter(
         }
       case s.MatchType(scrutinee, cases) =>
         s"${printType(scrutinee)} match { ${cases.size} cases }"
+      case s.LambdaType(scope, returnType) =>
+        val params = scope.map(_.symlinks).getOrElse(Nil)
```
Comment (tanishiking) [inline metals/src/main/scala/scala/meta/internal/decorations/SemanticdbTreePrinter.scala:90]: we shouldn't ignore hard links (actually, Scala3 links them by hardlinking)

```suggestion
        val params = scope.map(scopeInfo).getOrElse(Nil).map(_.displayName)
```

with

```scala
  private def scopeInfo(scope: s.Scope): List[s.SymbolInformation] =
    if (scope.symlinks.nonEmpty)
      scope.symlinks.map(symbol => s.SymbolInformation(symbol = symbol)).toList
    else
      scope.hardlinks.toList
```

**Review by tanishiking — APPROVED** [review]
I left one comment in LambdaType printer stuff, but otherwise LGTM!

File: metals/src/main/scala/scala/meta/internal/decorations/SemanticdbTreePrinter.scala (hunk 2 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/decorations/SemanticdbTreePrinter.scala]: Thanks! Changed!

File: sbt-metals/src/sbt-test/sbt-metals/semanticdb/build.sbt (hunk 1 above)
Comment (tgodzik) [reply sbt-metals/src/sbt-test/sbt-metals/semanticdb/build.sbt]: I rebased on the update PR.

**Review by tanishiking — APPROVED** [review]
thanks! LGTM

**Pull request merged**